### PR TITLE
feat(admin): add admin extension

### PR DESCRIPTION
**Pull request by xujianhai666**
## What is the purpose of the change

add admin extension

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,9 +12,12 @@ require (
 	github.com/tidwall/match v1.0.1 // indirect
 	github.com/tidwall/pretty v0.0.0-20190325153808-1166b9ac2b65 // indirect
 	go.uber.org/atomic v1.5.1
+	golang.org/x/sync v0.0.0-20190423024810-112230192c58
 	stathat.com/c/consistent v1.0.0
 )
 
 replace stathat.com/c/consistent v1.0.0 => github.com/stathat/consistent v1.0.0
 
+replace github.com/json-iterator/go => github.com/xujianhai666/go v1.1.10-0.20200118093701-ae604587eee1
```
Comment (ShannonDing) [inline go.mod:21]: remove personal pkg.

Comment (xujianhai666) [reply go.mod:21]: I should check the json-iterator has accept my work on this project, then I will remove this

Comment (ShannonDing) [reply go.mod:21]: this pr looks good to me, I will remove this personal pkg first, and merge this PR to the master as the initial code of the admin.

Comment (xujianhai666) [reply go.mod]: if remove replace, the code will crash @ShannonDing

Comment (ShannonDing) [reply go.mod]: the function "func (iter *Iterator) NextToken() byte" was not export in json-iterator/go, shall we request a PR for it?

Comment (ShannonDing) [reply go.mod]: how about "func (iter *Iterator) WhatIsNext() ValueType" ?

**Comment by ShannonDing** [comment]
link https://github.com/json-iterator/go/pull/442

File: admin/protocol.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	Language   remote.LanguageCode `json:"language"`
+	Version    int                 `json:"version"`
+}
+
+// ConsumerConnection consumer connection info fetched from broker, consumer register info by heartbeat
+type ConsumerConnection struct {
+	connectionSet    map[Connection]struct{}
+	consumeType      consumer.ConsumeType
+	messageModel     consumer.MessageModel
+	consumeFromWhere consumer.ConsumeFromWhere
```
Comment (liangyuanpeng) [inline admin/protocol.go:37]: Use pointer will better on struct

**Comment by Yiyiyimu** [comment]
Hi @ShannonDing If I make it correct, is this PR not used to support topic creation and deletion?

**Comment by Yiyiyimu** [comment]
Hi @xujianhai666 are you still working on this pr?

File: admin/admin.go [hunk 3]
```diff
@@ -0,0 +1,753 @@
...
+	cmd := remote.NewRemotingCommand(internal.ReqGetConsumerRunningInfo, req, nil)
+	response, err := a.cli.InvokeSync(ctx, brokerAddr, cmd, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	switch response.Code {
+	case internal.ResSuccess:
+		info := &internal.ConsumerRunningInfo{}
+		err := info.Decode(response.Body)
```
Comment (Yiyiyimu) [inline admin/admin.go:734]: Could we use `Unmarshal` instead of `Decode` like in line 663? It seems after the change, test could still pass.

**Pull request closed without merging**